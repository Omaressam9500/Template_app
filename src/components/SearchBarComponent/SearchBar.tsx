import { t } from 'i18next';
import React, { FC, useRef, useState } from 'react';
import {

    TextInput,
    View,
} from 'react-native';
import { styles } from './style';



interface ISearchHeader {
    handleSearchValue: (searchValue: string) => void;

}
const SearchBar: FC<ISearchHeader> = ({ handleSearchValue
}) => {
    const { container, textInput, textInputStyle } = styles()
    const [searchTxt, setSearchTxt] = useState('')
    return (
        <View style={[container]}>
            <View style={textInputStyle}>
                <TextInput style={textInput}
                    testID={"input"}
                    placeholder={t("Search")}
                    value={searchTxt}
                    placeholderTextColor={"#919C9C"}
                    returnKeyType="search"
                    onChangeText={(text) => {
                        setSearchTxt(text)
                        handleSearchValue(text)
                    }}


                    focusable={true}
                />
            </View>


        </View>


    );
};




export default SearchBar;
