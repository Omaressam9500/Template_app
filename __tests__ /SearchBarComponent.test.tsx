import React from 'react';
import { it, expect, describe, jest } from '@jest/globals';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import SearchBar from '../src/components/SearchBarComponent/SearchBar';


describe('SearchBarComponent', () => {
    it('Render ComponentCorrect', () => {
        const handleSearchValue = jest.fn()

        const tree = renderer.create(
            <SearchBar handleSearchValue={handleSearchValue} />,
        )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Should apply the value when changing text', () => {
        const handleSearchValue = jest.fn()
        const { getByTestId } = render(<SearchBar handleSearchValue={handleSearchValue} />);
        const input = getByTestId('input');
        fireEvent.changeText(input, "test");
        expect(input.props.value).toBe('test');
        expect(handleSearchValue).toHaveBeenCalledWith('test');
      });
})
