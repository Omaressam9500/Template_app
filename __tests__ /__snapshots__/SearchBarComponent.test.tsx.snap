// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarComponent Render ComponentCorrect 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "marginTop": 25,
        "paddingBottom": 17,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F9F9FF",
        "borderColor": "#000000",
        "borderRadius": 94,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "fontSize": 23,
        "height": 82,
        "marginRight": 12,
        "paddingLeft": 25,
        "paddingRight": 17,
      }
    }
  >
    <TextInput
      focusable={true}
      onChangeText={[Function]}
      placeholderTextColor="#919C9C"
      returnKeyType="search"
      style={
        {
          "color": "#000000",
          "flex": 1,
          "fontSize": 23,
          "textAlign": "left",
        }
      }
      testID="input"
      value=""
    />
  </View>
</View>
`;
